test/state/host.cpp

```diff
@@ -271,13 +271,22 @@ evmc::Result Host::create(const evmc_message& msg) noexcept
                                           evmc::Result{EVMC_FAILURE};
     }
 
-    if (m_rev >= EVMC_PRAGUE && (is_eof_container(initcode) || is_eof_container(code)))
+    if (!code.empty() && code[0] == 0xEF)
     {
-        if (validate_eof(m_rev, code) != EOFValidationError::success)
+        if (m_rev >= EVMC_LONDON && m_rev < EVMC_PRAGUE)
+        {
+            // EIP-3541: Reject EF code.
             return evmc::Result{EVMC_CONTRACT_VALIDATION_FAILURE};
+        }
+        else if (m_rev >= EVMC_PRAGUE)
+        {
+            // Only CREATE3/4 is allowed to deploy code starting with EF.
+            // It must be valid EOF, which was validated before execution.
+            if (msg.kind == EVMC_CREATE || msg.kind == EVMC_CREATE2)
```

Comment:
It will be always true on current master, but anticipates support for CREATE3/4

Reply:
Maybe use assert for now.